File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.cpp
class Solution {
public:
    bool uniformArray(vector<int>& nums1) {
        int mn = *min_element(nums1.begin(), nums1.end());

        // If the minimum element is odd,
        // every element can be made odd.
        if (mn % 2 == 1)
            return true;

        // If minimum is even, an odd element cannot be changed
        // because there is no smaller odd element for the first odd.
        for (int x : nums1) {
            if (x % 2 == 1)
                return false;
        }

        // All elements are even.
        return true;
    }
};

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.py
class Solution:
    def uniformArray(self, nums1):
        mn = min(nums1)

        # If minimum is odd, all elements can be made odd
        if mn % 2 == 1:
            return True

        # Minimum is even, so all elements must already be even
        for x in nums1:
            if x % 2 == 1:
                return False

        return True